src/components/flame-graph/Canvas.js

```diff
@@ -209,12 +209,24 @@ export function getHoverBackgroundColor(stackType: StackType): string {
 
 class FlameGraphCanvas extends React.PureComponent<Props> {
   _textMeasurement: null | TextMeasurement;
+  _chartCanvas: null | ChartCanvas<HoveredStackTiming>;
+  _setChartCanvasRef = chartCanvas => (this._chartCanvas = chartCanvas);
 
   constructor(props: Props) {
     super(props);
     (this: any)._getHoveredStackInfo = this._getHoveredStackInfo.bind(this);
     (this: any)._drawCanvas = this._drawCanvas.bind(this);
     (this: any)._hitTest = this._hitTest.bind(this);
+    this._chartCanvas = null;
+  }
+
+  componentWillReceiveProps(newProps: Props) {
+    if (
+      this._chartCanvas !== null &&
+      newProps.flameGraphTiming !== this.props.flameGraphTiming
+    ) {
+      this._chartCanvas.clearHoveredItem();
+    }
```

Comment:
Do you think this should be done in other Canvases too? Namely the stack chart and the marker chart. Or probably the stack chart only, as the marker chart isn't changed by symbolication.

I don't clearly understand why the flame graph is different.